zerver/lib/export.py

```diff
@@ -81,11 +81,7 @@
 IdSource = Tuple[TableName, Field]
 SourceFilter = Callable[[Record], bool]
 
-# This next type is a callback, which mypy does not
-# support well, because PEP 484 says "using callbacks
-# with keyword arguments is not perceived as a common use case."
-# CustomFetch = Callable[[TableData, Config, Context], None]
-CustomFetch = Any  # TODO: make more specific, see above
+CustomFetch = Callable[[TableData, "Config", Context], None]
```

Comment:
FTR, mypy now lets you write [callback protocols](https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols) with keyword arguments.

```python
class CustomFetch(Protocol):
    def __call__(self, response: TableData, config: "Config", context: Context) -> None: ...
```

Reply:
Good to know, thanks.